content/en/docs/languages/php/context.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Context
+weight: 55
+cSpell:ignore: Swoole
+description: Learn how context works in instrumented applications.
+---
+
+OpenTelemetry works by storing and propagating telemetry data. For example, when
+an instrumented application receives a request and a span starts, the span must
+be available to a component which creates child spans. To address this need,
+OpenTelemetry stores the span in the context.
+
+See the [Context specification](/docs/specs/otel/context/) for more information
+on what the context is and does.
+
+## Context
+
+Context is globally available within a single PHP execution context, and there
+can only be one [active context](#active-context) in the current execution
+context. Context can store values (for example, a `Span`).
+
+### Storage
+
+Context uses `Storage` to keep track of values. By default, a generic
+`ContextStorage` is used. OpenTelemetry for PHP supports other context storage
+for less common use cases, like asynchronous or concurrent execution with
+`fibers`.
+
+## Context keys
+
+Context entries are key-value pairs. Keys can be created by calling
+`OpenTelemetry\Context\Context::createKey()`.
+
+```php
+use OpenTelemetry\Context\Context;
+
+$key1 = Context::createKey('My first key');
+$key2 = Context::createKey('My second key');
+```
+
+Context keys are used to store and retrieve values from context.
+
+## Basic operations
+
+### Store and retrieve values
+
+Values are stored in Context by using the `$context->with($key, $value)` method.
+Setting a context entry creates a new context with the new entry in its storage,
+containing `$value`.
+
+Context is immutable, and setting a context entry does not modify the previous
+context. Once a value is stored in a context, it can be retrieved via
+`$context->get($key)`:
+
+```php
+use OpenTelemetry\Context\Context;
+
+$key = Context::createKey('some key');
+
+// add a new entry
+$ctx2 = Context::getCurrent()->with($key, 'context 2');
+
+// ctx2 contains the new entry
+var_dump($ctx2->get($key)); // "context 2"
+
+// active context is unchanged
+var_dump(Context::getCurrent()->get($key)); // NULL
+```
+
+If a value is not found in the current context, then each parent is checked
+until either the key is found, or the root context is reached.
+
+## Active context
+
+The active context is the context which is returned by `Context::getCurrent()`.
+The context object contains entries which allow telemetry components to
+communicate with each other. For example, when a span is created it can be
+activated, which creates a new active context and stores the span. Later, when
+another span is created it can use the span from the active context as its
+parent span. If no context is active, the root context is returned, which is
+just the empty context object.
```

Comment:
The reader was already exposed to `Context::getCurrent()` in the previous section, so I'd suggest the following section reorg:

> ## Context keys
> ## Active context
> ## Set and get context values
> ## Activate a context
> ## Scope
> ...
> ## Nested contexts
> ## Context in asynchronous environments

WDYT?

Reply:
> WDYT?

Done. I think it flows a lot better now, thanks.